be/src/exprs/quantile_function.cpp

```diff
@@ -0,0 +1,127 @@
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "exprs/quantile_function.h"
+#include "exprs/anyval_util.h"
+#include "gutil/strings/numbers.h"
+#include "gutil/strings/split.h"
+#include "util/quantile_state.h"
+#include "util/string_parser.hpp"
+#include "util/slice.h"
+
+namespace doris {
+
+using doris_udf::DoubleVal;
+using doris_udf::StringVal;
+using doris_udf::FloatVal;
+
+void QuantileStateFunctions::init(){}
+
+void QuantileStateFunctions::quantile_state_init(FunctionContext* ctx, StringVal* dst) {
+    dst->is_null = false;
+    dst->len = sizeof(QuantileState<double>);
+    dst->ptr = (uint8_t*) new QuantileState<double>();
+}
+
+static StringVal serialize(FunctionContext* ctx, QuantileState<double>* value) {
+    StringVal result(ctx, value->get_serialized_size());
+    value->serialize(result.ptr);
+    return result;
+}
+
+StringVal QuantileStateFunctions::to_quantile_state(FunctionContext* ctx, const StringVal& src) {
+    QuantileState<double> quantile_state;
+    quantile_state.set_compression(2048);
+    const AnyVal* digest_compression = ctx->get_constant_arg(1);
+    if (digest_compression != nullptr) {
+        float compression = reinterpret_cast<const FloatVal*>(digest_compression)->val;
+        if (compression >= 2048 && compression <= 10000) {
+            quantile_state.set_compression(compression);
+        }
+    }
+    
+    if(!src.is_null) {
+        StringParser::ParseResult parse_result = StringParser::PARSE_SUCCESS;
+        double double_value = StringParser::string_to_float<double>(
+                reinterpret_cast<char*>(src.ptr), src.len, &parse_result);
+        if (UNLIKELY(parse_result != StringParser::PARSE_SUCCESS)) {
+            std::stringstream error_msg;
+            error_msg << "The input: " << std::string(reinterpret_cast<char*>(src.ptr), src.len)
+                      << " is not valid, to_bitmap only support bigint value from 0 to "
+                         "18446744073709551615 currently";
+            ctx->set_error(error_msg.str().c_str());
+            return StringVal::null();
+        }
+        quantile_state.add_value(double_value);
+    }
+    return serialize(ctx, &quantile_state);
+}
+
+void QuantileStateFunctions::quantile_union(FunctionContext* ctx, const StringVal& src, StringVal* dst) {
+    if(src.is_null) {
+        return;
+    }
+    auto dst_quantile = reinterpret_cast<QuantileState<double>*>(dst->ptr);
+    if(src.len == 0) {
+        dst_quantile->merge(*reinterpret_cast<QuantileState<double>*>(src.ptr));
+    } else {
+        QuantileState<double> state(Slice(src.ptr, src.len));
+        dst_quantile->merge(state);
+    }
+}
+
+DoubleVal QuantileStateFunctions::quantile_percent(FunctionContext* ctx, StringVal& src) {
+    const AnyVal* percentile = ctx->get_constant_arg(1);
```

Comment:
I'm afraid we should add a `quantile_percent _prepare` function to do some preparation work, e.g., handle constant arguments checks.

Reply:
I did not get the benfit of adding a quantile_percent_prepare function. May be this can be implemented in support vec-functions of quantile_state.

Reply:
Please ignore the comments before.
Hmm... I think we should do arguments' validation check in FE, i.e., `FunctionCall.analyzeImpl`, rather than in BE.